### DISCUSSION TITLE
Model timestamp updated_at has default value set on creation

### DISCUSSION THREAD

**Discussion by cadavre** [strapi/strapi#5075] [Legacy Topics]
**Describe the bug**
`updated_at` field is auto-populated for Postgres because of column option to default to `CURRENT_TIMESTAMP`.

**Expected behavior**
`updated_at` field shall not be set on entity creation.

**Screenshots**
<img width="512" alt="Zrzut ekranu 2020-01-23 o 16 19 53" src="https://user-images.githubusercontent.com/461796/72997156-334da180-3dfc-11ea-878b-5e8f3e3a6980.png">

**Code snippets**
`car/models/Car.settings.json`:
```
{
  "info": {
    "name": "car"
  },
  "options": {
    "increments": true,
    "timestamps": [
      "created_at",
      "updated_at"
    ]
  },
  "attributes": {
    "name": {
      "type": "string"
    },
// ...
```

**System**
- Strapi version: 3.0.0-beta.18.6
- Database: Postgres

**Comment by lauriejim** [comment]
Hello! I'm not sure to follow you.
Can you please give me the details of your issue.

**Comment by cadavre** [comment]
On **creation** of an entity – `updated_at` field is updated along `created_at`. This is not what should be happening.

**Comment by lauriejim** [comment]
What should happen to you?

**Comment by cadavre** [comment]
Newly created entity (that had not beed updated yet), shall have `updated_at = null`.

Tons of the systems depends on this check: `updated_at IS NOT NULL` to check if an entity was changed anytime after creation.

**Comment by lauriejim** [comment]
Hum would like the @Aurelsicoko  and @alexandrebodin point of view about that.

**Comment by cadavre** [comment]
FYI it's how it was before `beta.18`. BTW same update since we cannot filter by datetime ranges in admin UI anymore. :(

**Comment by alexandrebodin** [comment]
Hi @cadavre this is pretty much a question of opinion here. We are not planning on changing this behaviour anytime soon. 

Having updated_at not being null is really practical in many usecases. You can just compare both timestamps if you really need to check if the entity was changed after creation.

**Reply by cadavre** [reply]
Unfortunately this is a very short-sighted attitude.

First of all you have especially two different fields **created**_at and **updated**_at. If you haven't updated an entity – updated_at shouldn't give you info that it was done the other way.

Another thing is atomicity in creating timestamps – and several [ms] difference between software creating a timestamp and database creating a timestamp, take a look at this:
<img width="363" alt="Zrzut ekranu 2020-04-17 o 11 18 57" src="https://user-images.githubusercontent.com/461796/79553877-7cc92800-809d-11ea-928c-f01075b65010.png">

> You can just compare both timestamps if you really need to check if the entity was changed after creation.

Looking at above – you can't.

Other topic is that I know many developers manually entering timestamps for created entities in their source code (and there's a good reason for that) before they're persisted into the database. This gives the same problem as above.

In short: **currently there's no way to check if entity was even updated or not**.

And `db.updated_at` being filled automatically if `entity.updated_at = null` – is a bug for me.

**Comment by derrickmehaffy** [comment]
Strapi doesn't directly set these values, the databases themselves do and this is pretty much the gold standard when it comes to SQL databases and automated `created_at` / `updated_at` columns.

Here is an article about it: https://medium.com/@bengarvey/use-an-updated-at-column-in-your-mysql-table-and-make-it-update-automatically-6bf010873e6a

And a discussion from Stack Overflow: https://softwareengineering.stackexchange.com/questions/371570/database-auditing-updated-at-null-or-not-null
